### DISCUSSION TITLE
Any way to turn throttling retries off?

### DISCUSSION THREAD

**Discussion by Amwins-Vault-Boy** [pnp/pnpcore#1542] [Q&A]
I'm going to look at the EventHub, but is there a configuration setting that would allow the automatic retry mechanism to be turned off in favor of just throwing the 429 exception from SP?

**Comment by jansenbe** [comment]
@Amwins-Vault-Boy : https://pnp.github.io/pnpcore/using-the-sdk/basics-settings.html documents the related options. Would recommend with trying to set UseRetryAfterHeader to false, MaxRetries to 0 and DelayInSeconds to 0. See https://github.com/pnp/pnpcore/blob/dev/src/sdk/PnP.Core/Services/Core/Http/RetryHandlerBase.cs.

Also, the Microsoft 365 Assessment tool uses EventHub, might be a good sample? https://github.com/pnp/pnpassessment/blob/main/src/PnP.Scanning/PnP.Scanning.Core/Services/ScanManager.cs#L40-L53
